arangod/VocBase/VocbaseInfo.cpp

```diff
@@ -357,6 +352,29 @@ VocbaseOptions getVocbaseOptions(ArangodServer& server, VPackSlice options) {
       vocbaseOptions.replicationFactor =
           replicationSlice
               .getNumber<decltype(vocbaseOptions.replicationFactor)>();
+      if (haveCluster) {
+        auto const minReplicationFactor =
+            server.getFeature<ClusterFeature>().minReplicationFactor();
+        auto const maxReplicationFactor =
+            server.getFeature<ClusterFeature>().maxReplicationFactor();
+        // make sure the replicationFactor value is between the configured min
+        // and max values
+        if (0 < maxReplicationFactor &&
+            maxReplicationFactor < vocbaseOptions.replicationFactor) {
+          THROW_ARANGO_EXCEPTION_MESSAGE(
+              TRI_ERROR_BAD_PARAMETER,
+              absl::StrCat("replicationFactor must not be higher than "
```

Comment:
Is StringUtils::concatT now deprecated in favor of absl or are we going to have both?

Reply:
Good question. I would favor using abseil because it is maintained for us.

Reply:
Also StringUtils::concatT isn't same.
It cannot convert numbers to string.
And few other issues

Reply:
what issues though? we should fix them, it is used in all active branches.

Reply:
I didn't mean bugs, so don't think necessary to change concatT code.

If you interested you can compare them.

Reply:
also it does convert number and any other type using `to_string` via adl.

Reply:
Good, forgot about to_string

Do you have some issue with abseil implementation? 

If you interested to use concat with custom strinify it's also possible

Reply:
I think my position about concanT same as Jan.

It's ok, but I don't see the reason to maintain own implementation of this function.

For example StrCat can use resize without memse for grow string, but now it's only for libc++.

Of course we can write our own better implementation which can do it for libstdc++ (similar code exist in folly), but is it necessary? I don't think so, but on 3rdParty library works a lot of people, and maybe sometime we update it and it will be done (issue already was created by someone).